libs/libskynet/src/types.ts

```diff
@@ -0,0 +1,68 @@
+// dataFn can take any object as input and has no return value. The input is
+// allowed to be undefined.
+type dataFn = (data?: any) => void;
+
+// error is an error type that is either a string or a null. If the value is
+// null, that indicates that there was no error. If the value is a string, it
+// indicates that there was an error and the string contains the error message.
+//
+// The skynet libraries prefer this error type to the standard Error type
```

Comment:
I'm wondering if passing a call stack along with the error message would be helpful?

If so, maybe we should actually construct `Error`s, so that we can then serialize them for `postMessage` calls - passing both the message and the callstack (available at `Error#stack`). Does not need to be addressed here, just something to consider for the future imho.